### PR TITLE
2006Scape Launcher

### PR DESCRIPTION
**Pull request by Dark98**
2006Scape Launcher is a Client/Bot Client Launcher For The 2006Scape Live Servers

### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com._2006Scape.Launcher.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>com._2006Scape.Launcher</id>
+  <launchable type="desktop-id">com._2006Scape.Launcher.desktop</launchable>
+  <name>2006Scape</name>
+  <developer_name>2006Scape</developer_name>
+  <summary>Launcher for 2006Scape</summary>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>AGPL-3.0-or-later</project_license>
+  <url type="homepage">https://2006Scape.org/</url>
```
Comment (hfiguiere) [inline com._2006Scape.Launcher.appdata.xml:10]: this is .org but the app-id uses .com. And they are not the same. What gives?

**Comment by hfiguiere** [comment]
bot, build com._2006Scape.Launcher

File: com._2006Scape.Launcher.appdata.xml (hunk 1 above)
Comment (Dark98) [reply com._2006Scape.Launcher.appdata.xml:10]: using com. was a mistake

File: com._2006Scape.Launcher.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  buildsystem: simple
+  build-commands:
+  - "/usr/lib/sdk/openjdk8/install.sh"
+- name: 2006Scape
+  buildsystem: simple
+  build-commands:
+  - install -D run.sh /app/bin/run.sh
+  - install -D 2006Scape_Launcher.jar /app/bin/2006Scape_Launcher.jar
+  - install -D com._2006Scape.Launcher.desktop /app/share/applications/com._2006Scape.Launcher.desktop
+  - install -D icon.svg /app/share/icons/hicolor/scalable/apps/com._2006Scape.Launcher.svg
```
Comment (hfiguiere) [inline com._2006Scape.Launcher.yml:19]: the appstream file is never installed.

**Comment by Dark98** [comment]
> Build [27909 failed](https://buildbot.flathub.org/#/builders/6/builds/27909)

is this an issue my end? (it builds fine for me, but this is my first flatpak so i'm not 100% what im doing)

**Comment by hfiguiere** [comment]
What's the relation between this and https://github.com/flathub/org._2009scape.Launcher

**Comment by Dark98** [comment]
> What's the relation between this and https://github.com/flathub/org._2009scape.Launcher

Their Both different Emulation Revsions of a game, ones based on the game in 2006 & the other is 2009

File: com._2006Scape.Launcher.yml (hunk 2 above)
Comment (Dark98) [reply com._2006Scape.Launcher.yml]: did you mean the appdata file?

**Comment by Dark98** [comment]
bot, build org._2006Scape.Launcher

**Comment by Dark98** [comment]
bot, build org._2006Scape.Launcher

File: org._2006Scape.Launcher.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  - type: file
+    path: icon.svg
+  - type: file
+    path: org._2006Scape.Launcher.appdata.xml
+    
+finish-args:
+  - --socket=fallback-x11
+  - --share=ipc
+  - --share=network
+  - --filesystem=home
```
Comment (barthalion) [inline org._2006Scape.Launcher.yml:37]: Why does it need full access to home filesystem?

File: org._2006Scape.Launcher.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  - install -D run.sh /app/bin/run.sh
+  - install -D 2006Scape_Launcher.jar /app/bin/2006Scape_Launcher.jar
+  - install -D org._2006Scape.Launcher.desktop /app/share/applications/org._2006Scape.Launcher.desktop
+  - install -D icon.svg /app/share/icons/hicolor/scalable/apps/org._2006Scape.Launcher.svg
+  - install -D org._2006Scape.Launcher.appdata.xml /app/share/metainfo/org._2006Scape.Launcher.appdata.xml
+  sources:
+  - type: file
+    path: run.sh
+  - type: file
+    path: 2006Scape_Launcher.jar
```
Comment (barthalion) [inline org._2006Scape.Launcher.yml:25]: At least this file should come from external source. GitHub is not a hosting for binary files.

File: org._2006Scape.Launcher.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  - type: file
+    path: org._2006Scape.Launcher.desktop
+  - type: file
+    path: icon.svg
+  - type: file
+    path: org._2006Scape.Launcher.appdata.xml
+    
+finish-args:
```
Comment (barthalion) [inline org._2006Scape.Launcher.yml:33]: `finish-args` are typically put before `modules`

File: org._2006Scape.Launcher.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  - --socket=fallback-x11
```
Comment (barthalion) [inline org._2006Scape.Launcher.yml:34]: You didn't specify `wayland`, so it likely should be `x11` instead.

**Comment by Dark98** [comment]
@barthalion Resolved all Review Changes :)

File: org._2006Scape.Launcher.yml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+command: run.sh
+sdk-extensions:
+- org.freedesktop.Sdk.Extension.openjdk8
+finish-args:
+  - --socket=x11
+  - --share=ipc
+  - --share=network
+  - --filesystem=~/2006Scape:create
```
Comment (hfiguiere) [inline org._2006Scape.Launcher.yml:12]: I think you should use persist instead.

File: org._2006Scape.Launcher.yml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  - install -D run.sh /app/bin/run.sh
+  - install -D 2006Scape_Launcher.jar /app/bin/2006Scape_Launcher.jar
+  - install -D org._2006Scape.Launcher.desktop /app/share/applications/org._2006Scape.Launcher.desktop
+  - install -D icon.svg /app/share/icons/hicolor/scalable/apps/org._2006Scape.Launcher.svg
+  - install -D org._2006Scape.Launcher.appdata.xml /app/share/metainfo/org._2006Scape.Launcher.appdata.xml
+  sources:
+  - type: file
+    path: run.sh
+  - type: file
+    url: https://2006Scape.org/Downloads/2006Scape_Launcher.jar
```
Comment (hfiguiere) [inline org._2006Scape.Launcher.yml:31]: it's not versionned? What happen when the next version is out?

File: org._2006Scape.Launcher.yml (hunk 7 above)
Comment (Dark98) [reply org._2006Scape.Launcher.yml:12]: We need manual access to ~/2006Scape to be able to edit the files inside & add Scripts used by the program

File: org._2006Scape.Launcher.yml (hunk 8 above)
Comment (Dark98) [reply org._2006Scape.Launcher.yml:31]: Does the sha256 not handle this?

Comment (hfiguiere) [reply org._2006Scape.Launcher.yml:31]: it will break the build for sure, which is why this should be versioned too.

Comment (Dark98) [reply org._2006Scape.Launcher.yml:31]: how do you mean it should be versioned? i'll just update the sha256 each time i update the file "https://2006Scape.org/Downloads/2006Scape_Launcher.jar"

**Comment by barthalion** [comment]
bot, build org._2006Scape.Launcher

**Pull request closed without merging**